llama_index/schema.py

```diff
@@ -44,63 +85,125 @@ class BaseDocument(DataClassJsonMixin):
     This must be a flat dictionary, 
     and only uses str keys, and (str, int, float) values.
     """
-    extra_info: Optional[Dict[str, Any]] = None
+    metadata: Dict[str, Any] = Field(
+        default_factory=dict, description="A flat dictionary of metadata fields"
+    )
+    usable_metadata: Optional[List[str]] = Field(
```

Comment:
nit: usable sounds a bit vague. Any other naming that's more explicit?

Reply:
Hmmm, maybe:
- `allowed_metadata_keys`
- `useable_metadata_keys`
- `injected_metadata_keys`
- `readable_metadata_keys`
- `metadata_keys_to_use`

Reply:
I also wonder, if we should give separate controls for which metadata is included when calculating embedding, vs included as content when shoving into context window.

Reply:
Is that too much complexity for a user? I was thinking the same thing though